sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -217,6 +396,10 @@ object RapidsBufferCatalog extends Logging with Arm {
   }
 
   private def closeImpl(): Unit = {
+    if (singleton != null) {
+      singleton.close()
```

Comment:
nit: I am confused by this. singleton is a `val` and it is set at startup. When will it ever be null?